### PR TITLE
udisks2 - 2.8.4 -> 2.9.4

### PR DESCRIPTION
**Pull request by kshlm**
###### Motivation for this change

Update to a version that supports the ntfs3 kernel module.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/udisks/2-default.nix [hunk 1]
```diff
@@ -37,17 +37,11 @@ stdenv.mkDerivation rec {
         xfsprogs ntfs3g parted util-linux
       ];
     })
-
-    # Fix tests: https://github.com/storaged-project/udisks/issues/724
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/udisks/2-default.nix:42]: Please remove the unused fetchpatch from inputs.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 147606` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages marked as broken and skipped:
</summary>
<ul>
<li>gnome.gnome-documents</li>
<li>xmonad_log_applet</li>
</ul>
</details>
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>gnome.gnome-music: log was empty</li>
  </ul>
</details>
<details>
<summary>
114 packages built:
</summary>
<ul>
<li>CuboCore.coreaction</li>
<li>CuboCore.corearchiver</li>
<li>CuboCore.corefm</li>
<li>CuboCore.coregarage</li>
<li>CuboCore.corehunt</li>
<li>CuboCore.coreimage</li>
<li>CuboCore.coreinfo</li>
<li>CuboCore.corekeyboard</li>
<li>CuboCore.corepad</li>
<li>CuboCore.corepaint</li>
<li>CuboCore.corepdf</li>
<li>CuboCore.corepins</li>
<li>CuboCore.corerenamer</li>
<li>CuboCore.coreshot</li>
<li>CuboCore.corestats</li>
<li>CuboCore.corestuff</li>
<li>CuboCore.coreterminal</li>
<li>CuboCore.coretime</li>
<li>CuboCore.coretoppings</li>
<li>CuboCore.coreuniverse</li>
<li>CuboCore.libcsys</li>
<li>adapta-gtk-theme</li>
<li>almanah</li>
<li>bubblemail</li>
<li>calls</li>
<li>cantata</li>
<li>chatty</li>
<li>chrome-gnome-shell</li>
<li>cinnamon.cinnamon-common</li>
<li>cinnamon.cinnamon-control-center</li>
<li>cinnamon.cinnamon-gsettings-overrides</li>
<li>cinnamon.cinnamon-screensaver</li>
<li>clementine</li>
<li>clementineUnfree</li>
<li>dropbox-cli</li>
<li>elementary-planner</li>
<li>empathy</li>
<li>enlightenment.enlightenment</li>
<li>evolution</li>
<li>evolution-data-server</li>
<li>evolution-ews</li>
<li>evolutionWithPlugins</li>
<li>exaile</li>
<li>far2l</li>
<li>feedreader</li>
<li>folks</li>
<li>gfbgraph</li>
<li>gnome-multi-writer</li>
<li>gnome-online-accounts</li>
<li>gnome-photos</li>
<li>gnome-recipes</li>
<li>gnome.cheese</li>
<li>gnome.file-roller</li>
<li>gnome.geary</li>
<li>gnome.gnome-applets</li>
<li>gnome.gnome-books</li>
<li>gnome.gnome-boxes</li>
<li>gnome.gnome-calendar</li>
<li>gnome.gnome-contacts</li>
<li>gnome.gnome-control-center</li>
<li>gnome.gnome-disk-utility</li>
<li>gnome.gnome-flashback</li>
<li>gnome.gnome-initial-setup</li>
<li>gnome.gnome-maps</li>
<li>gnome.gnome-notes</li>
<li>gnome.gnome-online-miners</li>
<li>gnome.gnome-panel</li>
<li>gnome.gnome-session</li>
<li>gnome.gnome-shell</li>
<li>gnome.gnome-terminal</li>
<li>gnome.gnome-todo</li>
<li>gnome.gnome-tweaks</li>
<li>gnome.gnome-user-share</li>
<li>gnome.gvfs</li>
<li>gnome.nautilus</li>
<li>gnome.nautilus-python</li>
<li>gnome.totem</li>
<li>gnomeExtensions.easyScreenCast</li>
<li>gnomeExtensions.gsconnect</li>
<li>gnomeExtensions.night-theme-switcher</li>
<li>grilo-plugins</li>
<li>gvfs</li>
<li>kgx</li>
<li>libgdata</li>
<li>libzapojit</li>
<li>mailnagWithPlugins</li>
<li>mate.mate-utils</li>
<li>monitor</li>
<li>pantheon.elementary-calendar</li>
<li>pantheon.elementary-greeter</li>
<li>pantheon.elementary-mail</li>
<li>pantheon.elementary-session-settings</li>
<li>pantheon.elementary-tasks</li>
<li>pantheon.file-roller</li>
<li>pantheon.file-roller-contract</li>
<li>pantheon.switchboard-plug-onlineaccounts</li>
<li>pantheon.switchboard-with-plugs</li>
<li>pantheon.wingpanel-applications-menu</li>
<li>pantheon.wingpanel-indicator-datetime</li>
<li>pantheon.wingpanel-with-indicators</li>
<li>phosh</li>
<li>psensor</li>
<li>rabbitvcs</li>
<li>rapid-photo-downloader</li>
<li>shotwell</li>
<li>spaceFM</li>
<li>synology-drive</li>
<li>tracker-miners</li>
<li>udiskie</li>
<li>udisks</li>
<li>usermount</li>
<li>vifm-full</li>
<li>vimix-gtk-themes</li>
<li>whitesur-gtk-theme</li>
</ul>
</details>

File: pkgs/os-specific/linux/udisks/fix-paths.patch [hunk 2]
```diff
@@ -22,17 +26,37 @@
...
+ Type=oneshot
+ RemainAfterExit=no
+ EnvironmentFile=-@zramconfdir@/%i
+-ExecStart=-/bin/sh -c 'if [ -n "$ZRAM_NUM_STR" ]; then echo "$ZRAM_NUM_STR" > /sys/class/block/%i/max_comp_streams; fi'
+-ExecStart=-/bin/sh -c 'if [ -n "$ZRAM_DEV_SIZE" ]; then echo "$ZRAM_DEV_SIZE" > /sys/class/block/%i/disksize; fi'
+-ExecStart=-/bin/sh -c 'if [ "$SWAP" = "y" ]; then mkswap /dev/%i && swapon /dev/%i; fi'
+-# ExecStop=-/bin/sh -c 'echo 1 > /sys/class/block/%i/reset'
++ExecStart=-@sh@ -c 'if [ -n "$ZRAM_NUM_STR" ]; then echo "$ZRAM_NUM_STR" > /sys/class/block/%i/max_comp_streams; fi'
++ExecStart=-@sh@ -c 'if [ -n "$ZRAM_DEV_SIZE" ]; then echo "$ZRAM_DEV_SIZE" > /sys/class/block/%i/disksize; fi'
++ExecStart=-@sh@ -c 'if [ "$SWAP" = "y" ]; then @mkswap@ /dev/%i && @swapon@ /dev/%i; fi'
```
Comment (jtojnar) [inline pkgs/os-specific/linux/udisks/fix-paths.patch:43]: Looks like you forgot to add `mkswap` and `swapon` to `substituteAll`.

File: pkgs/os-specific/linux/udisks/fix-paths.patch [hunk 3]
```diff
@@ -41,18 +65,11 @@
  # vim: dict+=/usr/share/beakerlib/dictionary.vim cpt=.,w,b,u,t,i,k
  # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  #
+diff --git a/src/tests/integration-test b/src/tests/integration-test
+index 4499a6a9..8b711f95 100755
 --- a/src/tests/integration-test
 +++ b/src/tests/integration-test
-@@ -414,7 +414,7 @@
```
Comment (jtojnar) [inline pkgs/os-specific/linux/udisks/fix-paths.patch:46]: Why was this hunk removed? It still seems to apply.

**Comment by jtojnar** [comment]
I opened https://github.com/NixOS/nixpkgs/pull/163926 fixing that.

**Pull request merged**